### PR TITLE
[BugFix] Stop caching Tokenizer parent.device

### LINKED ISSUE #4345: [BUG] Tokenizer caches parent.device and goes stale after env.to
## Describe the bug

Both LLM `Tokenizer` and `torchrl.envs.transforms` `Tokenizer` cache `parent.device` on first access:

```python
@property
def device(self):
    if "_device" in self.__dict__:
        return self._device
    ...
    self._device = parent.device
    return device
```

`env.to(...)` / `transform.to(...)` moves parameters and the parent device, but not this Python attribute (Agents.md 6a / same class as #4224). Later tokenize calls still `.to()` onto the stale device.

## Expected behavior

Do not cache a single module device. Read the parent (or the input tensor) on each call. After `env.to("meta")`, new `input_ids` must be on `meta`.

## Checklist

- [x] I have checked that there is no similar issue in the repo (**required**)
- [x] I have read the documentation (**required**)
- [x] I have provided a minimal working example to reproduce the bug (**required**)

### PR DESCRIPTION
**Pull request by YeonwooSung**
## Description

Both LLM `Tokenizer` and `torchrl.envs.transforms.Tokenizer` cached `parent.device` on first access. `env.to(...)` does not move that Python attribute, so later tokenize calls stayed on the stale device.

`out_device` now reads `parent.device` on each call and describes where token tensors and attention masks are written. After `env.to("meta")`, both outputs are on `meta`. When `out_device` is `None`, outputs keep the device chosen by the tokenizer.

Both Tokenizer implementations retain `device` as a deprecated alias, with removal scheduled for TorchRL v0.17. Reading the alias emits a `DeprecationWarning`; normal tokenization uses `out_device` and does not warn.

### Code example

Use `out_device` to inspect tokenization output placement. Tokenization follows an environment device move even after the first call. The small tokenizer stub avoids a model download; the same API is available on `torchrl.envs.transforms.Tokenizer`.

```python
import torch
from tensordict import TensorDict
from torchrl.envs import TransformedEnv
from torchrl.envs.llm.transforms import Tokenizer
from torchrl.testing.mocking_classes import CountingEnv

class FixedTokenizer:
    def __call__(self, value, **kwargs):
        return {
            "input_ids": torch.tensor([[1, 2]]),
            "attention_mask": torch.tensor([[1, 1]]),
        }

transform = Tokenizer(
    in_keys=["text"], out_keys=["input_ids"], tokenizer=FixedTokenizer(),
)
env = TransformedEnv(CountingEnv(), transform)
assert transform(TensorDict(text="hello"))["input_ids"].device.type == "cpu"
env.to("meta")
assert transform.out_device == torch.device("meta")
assert transform(TensorDict(text="hello"))["input_ids"].device.type == "meta"
# The second result used to stay on CPU because parent.device was cached.
env.close()
```

## Motivation and Context

close #4345

Same class of bug as #4224.

- [x] I have raised an issue to propose this change ([required](https://github.com/pytorch/rl/issues) for new features and bug fixes)

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [x] I have read the [CONTRIBUTION](https://github.com/pytorch/rl/blob/main/CONTRIBUTING.md) guide (**required**)
- [x] My change requires a change to the documentation.
- [x] I have updated the tests accordingly (*required for a bug fix or a new feature*).
- [x] I have updated the documentation accordingly.

### CODE REVIEW

**Review by vmoens — APPROVED** [review]
The property now follows the live parent device in both Tokenizer implementations. Both device-move regressions passed locally. No actionable findings. GitHub CI has not run yet; the workflows are awaiting approval.

**Review by vmoens — COMMENTED** [review]
The live parent lookup fixes the stale output placement correctly. Naming recommendation: expose this as `out_device`, since it describes where tokenization results are written, rather than claiming all module state lives on one device.

Keep the existing `device` property as a deprecated alias until TorchRL v0.17, and use `out_device` internally so normal tokenization does not emit deprecation warnings. Apply the same naming to both Tokenizer implementations. The replacement property block would be:

```python
# Module-level import:
import warnings

# Inside Tokenizer:
@property
def out_device(self) -> torch.device | None:
    parent = self.parent
    return None if parent is None else parent.device

@property
def device(self) -> torch.device | None:
    warnings.warn(
        "Tokenizer.device is deprecated and will be removed in TorchRL v0.17. "
        "Use out_device instead.",
        DeprecationWarning,
        stacklevel=2,
    )
    return self.out_device
```

`None` should continue to mean no enforced destination. This is a naming/deprecation recommendation; the existing cache fix remains valid. It does not require adding a new constructor override or changing where outputs are placed.
